### PR TITLE
Add dict increment

### PR DESCRIPTION
**Pull request by kabra1110**
Towards #1804.
Does not work with `str` keys. Looked around `llvm_utils`, but not sure what the issue is.

### CODE REVIEW

File: integration_tests/test_dict_increment.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    s1: str
+
+    d_int_int = {0: 1}
+    d_int_int[0] += 1000
+    assert d_int_int[0] == 1001
+
+    i2 = 1
+    d_int_int = {1: i2}
+    for i1 in range(10):
+        d_int_int[1] += i1
```
Comment (czgdp1807) [inline integration_tests/test_dict_increment.py:20]: Also test with runtime keys. Right now all the keys in your test are compile time constants.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

**Comment by czgdp1807** [comment]
> Looked around `llvm_utils`, but not sure what the issue is.

What's the error message you are receiving?

**Comment by kabra1110** [comment]
> What's the error message you are receiving?

None. The value isn't incremented.
```
from lpython import i32

def test_dict_increment():
    d_str_int: dict[str, i32]
    d_str_int = {"1": 1}
    print(d_str_int["1"])
    d_str_int["1"] += 1
    print(d_str_int["1"])
    d_str_int["1"] = d_str_int["1"] + 1
    print(d_str_int["1"])

test_dict_increment()
```
```
1
1
2
```

File: integration_tests/test_dict_increment.py (hunk 1 above)
Comment (kabra1110) [reply integration_tests/test_dict_increment.py:22]: Added

**Comment by czgdp1807** [comment]
> ```python
> d_str_int["1"] += 1
> ```

This doesn't work.



> ```python
> d_str_int["1"] = d_str_int["1"] + 1
> ```

But this works. So why can't we implement `... += ...` as `... = ... + ...` in LLVM?

**Comment by czgdp1807** [comment]
Also we will have to see how this operation looks like in ASR. We will see in our next meeting.

File: integration_tests/test_dict_increment.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        if j1 < 2.0:
+            j2 += 0.1
+        j1 += 0.1
+    assert d_bool_float[j1 < 2.0] == d_bool_float[j1 > 2.0]
+    assert d_bool_float[True] == j2
+
+    j1 = 2.0
+    d_str_float = {'key': j1}
+    while j1 < 4.0:
+        d_str_float['ke' + 'y'] += 0.1
```
Comment (czgdp1807) [inline integration_tests/test_dict_increment.py:54]:
```suggestion
    d_str_float = {'key': j1}
    s1 = "ke"
    while j1 < 4.0:
        d_str_float[s1 + 'y'] += 0.1
```

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

**Comment by certik** [comment]
Looks like more work is needed. @kabra1110 go ahead and click on "Ready for review" once this is ready.

**Review by czgdp1807 — APPROVED** [review]

**Pull request merged**